eahub/profiles/views.py

```diff
@@ -173,3 +180,38 @@ def delete_profile(request):
         return render(request, 'eahub/delete_profile.html', {
             'form': form
         })
+
+class ReportAbuse:
+    def __init__(self, reportee, type):
+        self.reportee = reportee
+        self.type = type
+
+    def send(self, request):
+        reasons = request.POST.getlist('reasons')
+        if not reasons:
+            messages.error(
+                request,
+                ''' You need to select at least one reason. ''',
+            )
+        else:
+            subject = "EA {0} reported as abuse: {1}".format(self.type, self.reportee.name)
+            message = render_to_string('emails/report_{}_abuse.txt'.format(self.type), {
+                'profile_name': self.reportee.name,
+                'profile_url': "https://{0}/profile/{1}".format(get_current_site(request).domain,self.reportee.slug),
+                'reasons': ', '.join(reasons)
+            })
+            send_mail(subject, message, settings.DEFAULT_FROM_EMAIL, recipient_list=settings.LEAN_MANAGERS)
+            return redirect('/{0}/{1}/report-abuse-done'.format(self.type,self.reportee.slug))
+
+
+
+def report_abuse(request, slug):
```

Comment:
See above on `FormView`. Also, this code looks basically the same across both apps; maybe separate it out into a shared dependency?

Reply:
put this into formview